pkg/search/proxy/store/multi_cluster_cache.go

```diff
@@ -355,16 +400,39 @@ func (c *MultiClusterCache) Watch(ctx context.Context, gvr schema.GroupVersionRe
 		if err != nil {
 			return nil, err
 		}
-
-		mux.AddSource(w, func(e watch.Event) {
+		encodePb := c.cachedResources[gvr].encode && usePb
+		mux.AddSource(w, func(e watch.Event) watch.Event {
```

Comment:
I don't quite understand why the return value needs to be modified. The handling of resources should directly affect `e.Object`, am I right?

Reply:
watch event.object if is CacheableObject ,use pb will message does not implement the protobuf marshalling interface and cannot be encoded to a protobuf message.
so if use watch, should modify the object.